pkg/auth/dependency/provider/oauth/oauth.go

```diff
@@ -4,7 +4,8 @@ const providerName string = "oauth"
 
 type Provider interface {
 	GetPrincipalByProviderUserID(providerName string, providerUserID string) (*Principal, error)
-	GetPrincipalByUserID(userID string) (*Principal, error)
+	GetPrincipalByUserID(providerName string, userID string) (*Principal, error)
 	CreatePrincipal(principal Principal) error
 	UpdatePrincipal(principal *Principal) error
+	DeletePrincipal(providerName string, principal *Principal) error
```

Comment:
This looks a bit weird, coz the principal itself should have the providerName.

Reply:
@Steven-Chan I had updated this, please help review. Thanks.